lock-api-objects/src/main/java/com/palantir/lock/v2/AcquireNamedMinTimestampLeaseResult.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.v2;
+
+import com.palantir.timestamp.TimestampRange;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface AcquireNamedMinTimestampLeaseResult {
+    @Value.Parameter
+    long getMinLeased();
+
+    @Value.Parameter
+    LockToken getLock();
+
+    @Value.Parameter
+    TimestampRange getFreshTimestamps();
```

Comment:
Actually, I think I should make this a set of longs instead of a range. I don't think it will be easy to guarantee a certain range (even with retries).

Instead, I will allow the timelock API to return a smaller number than required. It will be then up to the glue code in between to retry until we have the number of timestamps we want.

Reply:
Yes, the API should guarantee you're going to get the number you asked for, otherwise it's just weird.

Reply:
Using `List<Long>`